pandas/core/dtypes/cast.py

```diff
@@ -1970,7 +1970,8 @@ def np_can_hold_element(dtype: np.dtype, element: Any) -> Any:
             if tipo.kind not in ["i", "u"]:
                 if isinstance(element, np.ndarray) and element.dtype.kind == "f":
                     # If all can be losslessly cast to integers, then we can hold them
-                    casted = element.astype(dtype)
+                    with np.errstate(invalid="ignore"):
```

Comment:
IIUC this is where the nan, 1.0 -> int cast is attempted? Could you add a comment to that affect here

Reply:
Yes correct, we are casting but we are checking afterwards if ot was losslessly, so warning does not offer any value

Reply:
Done